### DISCUSSION TITLE
230703 사용자 스토리 회의 ( 13:25  ~ ... )

### DISCUSSION THREAD

**Discussion by splitCoding** [woowacourse-teams/2023-shook#9] [회의]
## 회의 안건

서비스의 사용자 스토리를 완성한다.

### 최종 시나리오 
'인디음악 매니아'는 자신이 좋아하는 인디가수의 신곡을 사람들에게 알려주고 싶었다. 잘 안 알려진 음악이라 노래에서 가장 좋은 '킬링파트'를 공유하려 한다. 노래를 사람들과 공유하기 위해 S-HOOK에 접속하여 해당 노래를 검색했다. 노래에서 가장 좋은 부분을 선택해서 등록하고 공유했다. 친구가 추천해줬던 노래들을 다 들어보고 싶지만 업무가 바빠서 해당 노래를 검색하여 킬링파트만 들어봤다.

### 3차 회의 16:15 에 재시작

**Comment by Cyma-s** [comment]
인디음악 매니아는 노래를 듣다가 심금을 울리는 부분을 발견했다. 친구한테 노래 링크를 보내주고, '1분 30초부터 들으면 돼~' 라고 보냈다. 친구가 1분 30초부터 듣기 귀찮다고 해서, 유튜브의 클립 기능을 이용해 좋은 부분만 자른 링크를 다시 보내줬다. 다른 사람들도 내가 좋아하는 부분을 좋아하는지 궁금해서, 다른 사람들이 추천하는 킬링파트도 들어봤다. 내가 추천했던 부분과 똑같은 부분을 추천하는 사람들이 많아서 기분이 좋았다.

리스너는 반복 듣기하던 노래가 질려서 새로운 좋은 노래를 찾고 싶다. 마침 인디 음악 매니아에게 음악 추천을 받아서 추천 파트부터 들어봤는데, 취향인 노래라서 플레이 리스트에 추가했다.

**Comment by ukkodeveloper** [comment]
'인디음악 매니아'는 자신이 좋아하는 인디가수의 신곡을 사람들에게 알려주고 싶었다. 인디 음악이기 때문에 노래에서 가장 좋은 '킬링파트'를 공유하려한다. 노래를 공유하기 위해 S-HOOK에 접속하여 해당 노래를 검색한다. 해당 곡에 이미 킬링파트가 등록되어있는 지부터 확인했다. 아무래도 인디가수의 곡이라 등록된 킬링파트가 없었고, 그는 노래에서 가장 좋은 부분을 선택했다. 선택 후에 자신 지인들에게 링크를 공유했다.

**Comment by seokhwan-an** [comment]
### 인디음악 매니아
내가 좋아하는 음악의 한 부분을 타인(친구)에게 알려주고 싶다. 그것을 알려주기 위해 타인(친구)에게 음악과 함께 어디부터 들어야하는지를 알려주었다. 또한 내가 좋아하는 가수의 새 음악이 나왔는데 그 노래에서 가장 좋은 부분을 듣기 위해 노래를 검색했다. 검색한 노래에서 사람들이 추천한 부분부터 찾아서 듣고 그 음악의 전체 부분을 들어보았다.

**Comment by somsom13** [comment]
### 인디음악 마니아의 시나리오

인디음악 마니아는 자신이 좋아하는 노래를 듣다가, 감명 깊은 파트를 발견했다. 이 부분을 다른 사람들도 들어보고 같이 좋아해줬으면 좋겠다는 마음으로 우아한 플레이리스트 채널에 들어갔다. 영상의 링크와 함께 추천하는 파트의 시간을 공유했고 많은 크루들이 해당 메세지에 👍 , 👎  반응을 남겨주었다. 채널에 들어온 김에 다른 좋은 노래들을 찾기 위해 우플리에서 내가 좋아하는 가수를 검색했더니, 다른 크루들이 공유한 음악을 찾을 수 있었다. 해당 음악들에 다른 크루들이 올려준 추천 시간대를 들어보았다.

### 리스너의 시나리오

한 달 동안 하나의 음악만 반복해서 들은 리스너는 음악에 질리기 시작했다. 내 취향의 새로운 음악을 찾기 위해 유X브에 접속하여 가장 조회수가 많은 최근 음악 영상들을 확인했다. 가장 조회수가 높은 영상에서 “가장 많이 돌려본 부분”을 먼저 들어보았다. 이 노래는 자신의 취향이 아니라서 다음 영상으로 넘어가 “많이 돌려본 부분”들을 들어보았다.

**Comment by jariita** [comment]
### 인디음악 매니아

인디음악 매니아는 좋아하는 인디가수의 노래를 듣다가 어떤 한 부분에 확 꽂혔다. 친구들에게 공유하기 위해서 노래 링크를 보내며 꽂혔던 부분의 가사나 시간을 이야기 하면서 꼭 들어보라고 권유한다.

### 리스너

리스너는 유튜브에서 플레이리스트를 검색하거나 유튜브 뮤직이 추천해주는 음악을 들었다. 리스너가 좋아하는 가수의 노래로 타임라인을 옮기는 것도 귀찮았고, 노래를 듣다가도 마음에 들지 않아 다음 노래로 넘기는 경우도 많았다.

**Comment by splitCoding** [comment]
### 인디음악 매니아(러너)의 시나리오

러너는 유명하지 않은 노래를 듣다 너무 좋은 부분이 생겼다. 이 부분을 공유하고 싶어 S-HOOK 에 접속하여 해당  노래를 검색해 보았다. 러너는 사람들의 킬링파트가 어딘지도 확인하고 그가 생각하는 킬링파트도 등록했다. 러너는 노래 페이지에서 공유를 눌러 친구에게 공유했다. 이전에 친구가 추천해줬던 노래들을 다 들어보고 싶지만 시간이 넉넉하지 않아 해당 노래를 검색하여 킬링파트가 어딘지를 확인하였다.

**Comment by Creative-Lee** [comment]
### 사용자 스토리 : 인디음악 매니아 이씨는 인디 음악을 사랑하는 리스너로서 노래 공유 기능을 원한다.

### 인수 조건
#### 첫번째

- (given) 이씨가 특정 음악의 킬링파트 등록 페이지에서
- (given) 킬링파트를 지정한 후 
- (when) 등록 or ㅇㅈ투표 버튼을 누르면
- (then) 불특정 다수의 유저에게 공유된다.

#### 두번째
- (given) 이씨가 등록한 킬링파트를 확인할 수 있는 특정 페이지에서
- (when) 공유 버튼을 누르면
- (then) (ㅇㅈ투표 or 추가 등록기능을 포함한) (킬링파트 상세페이지로 가는?) 공유 링크가 생성된다.

**Comment by Cyma-s** [comment]
'인디음악 매니아'는 자신이 좋아하는 인디가수의 신곡을 사람들에게 알려주고 싶었다. 잘 안 알려진 음악이라 노래에서 가장 좋은 '킬링파트'를 공유하려 한다. 노래를 사람들과 공유하기 위해 S-HOOK에 접속하여 해당 노래를 검색했다. 노래에서 가장 좋은 부분을 선택해서 등록하고 공유했다. 친구가 추천해줬던 노래들을 다 들어보고 싶지만 업무가 바빠서 해당 노래를 검색하여 킬링파트만 들어봤다.

**Comment by Cyma-s** [comment]
## 사용자 스토리 + 인수조건

**1. 나는 인디음악 매니아로서 노래의 좋은 부분을 공유하기 위해 킬링파트를 등록하는 기능을 원한다.**

(given)
- 인디음악 매니아가 특정 음악의 킬링파트 등록 페이지에서
- 킬링파트를 지정한 후

(when)
- 등록 버튼을 누르면

(then)
- 특정 음악의 킬링파트가 등록된다


**2. 나는 바쁜 직장인으로서 추천받은 음악의 좋은 부분을 빨리 찾기 위해 킬링파트를 찾는 기능을 원한다.**

(given)
- 바쁜 직장인이 음악의 킬링파트 조회 페이지에 접속한 후

(then)
- 음악의 킬링파트를 조회할 수 있다
